app/services/backup_code_generator.rb

```diff
@@ -32,6 +32,18 @@ def verify(plaintext_code)
     true
   end
 
+  # TODO Note: This is a mirror of the style found in:
+  # app/forms/totp_verification_form.rb
+  # @return [BackupCodeConfiguration, nil]
+  def if_valid_code_return_config(plaintext_code)
+    return unless plaintext_code.present?
+    backup_code = RandomPhrase.normalize(plaintext_code)
+    config = BackupCodeConfiguration.find_with_code(code: backup_code, user_id: @user.id)
+    return unless code_usable?(config)
+    config.update!(used_at: Time.zone.now)
+    config
```

Comment:
To avoid duplicating this logic between here and in `verify`, can we refactor `verify` to use this method?

i.e.

```rb
def verify(plaintext_code)
  if_valid_code_return_config(plaintext_code).present?
end
```